sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -846,6 +846,17 @@ class Column(val expr: Expression) extends Logging {
    */
   def rlike(literal: String): Column = withExpr { RLike(expr, lit(literal).expr) }
 
+  /**
+   * SQL ILIKE expression (case insensitive LIKE).
+   *
+   * @group expr_ops
+   * @since 3.3.0
+   */
+  def ilike(literal: String, escapeChar: Char = '\\'): Column = withExpr {
```

Comment:
Let's forget about `escapeChar` for now. I don't think it's used pretty often, and it's not exposed in `like` above too.

Reply:
I think so too, but I'd like to confirm that from SQL, it seems we can use `escapeChar`.
https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala#L231
Is that OK to close `escapeChar` on dataframe ?

Reply:
yeah, let's don't add it for now. these DSL APIs don't usually have all matched APIs but just commonly used ones (that make sense to Scala, Python, etc users instead of SQL users)